libflux/flux-core/src/semantic/nodes.rs

```diff
@@ -1556,6 +1556,11 @@ impl MemberExpr {
             }
         }
 
+        if let MonoType::Dynamic(_) = &t {
+            self.typ = t.clone();
+            return Ok(());
+        }
+
```

Comment:
@onelson This and the accompanying change to `Expression::Index.infer()` were my solution to get the new type inference tests to pass. This seems... hacky tough. Open to suggestions for other solutions.

Reply:
I think that looks fair. I'm no expert though. My reading of this says that when `r` is dynamic, assign the type of dynamic to expression accessing the property (which is what we want).

The fact you're cloning `t` to do this is okay, I think? Otherwise you could look to assign a new instance of the type (but I'm not sure why that'd matter). There's a flatbuffer pointer inside which I'm not sure how to construct by hand :sob:

Someone else with more experience in this are should probably check this out. Maybe Jonathan has input.

Reply:
Doing more targeted checks like this seems better to me generally, but I could maybe see issues depending on the order that type inference runs. That is, this requires that we know that the type is `dynamic` when we reach this point in the code, we can't discover that it is dynamic later and then pass this check. I don't know if that is actually a problem though

Reply:
Did we ever figure this out? It seems to me like it's possible this condition might not work correctly if we haven't yet determined that the original object was dynamic.

Just to make sure, if it turns out this fails, we can just turn dynamic off and this section of code will just not matter, correct?

Reply:
I think the deal is all the fns that will construct `Dynamic` items will be in experimental, but iirc there's no reasonable way to feature flag this stuff.

It does follow that if user's don't construct dynamic items, this match arm won't matter.

Reply:
Building on what Owen said, nothing currently uses the dynamic type. The functions that will use it will come from the other issues in the epic. I think the patches for those issues will be easier to test.

Reply:
@jsternberg Curious about your thoughts on this. I don't want to merge this if you have serious doubts. However, this is currently blocking other work, so if there's a specific change I can make to get this PR across the line, I'm happy to make it.